### PR TITLE
[DO NOT MERGE] [REPRODUCING ERROR] Error with convtranspose 2d unit test

### PR DESCRIPTION
**Pull request by SrivastavaKshitij**
Hi @jaybdub 

I am getting the following error with one of the combinations of unit tests for ConvTranspose2D layer and I dont know if its a torch2trt bug or the bug in TRT itself. 

Error doesnt go away no matter how much workspace size is increased.

```
root@7f9f14f6c747:/workspace/work# vim torch2trt/torch2trt.py
root@7f9f14f6c747:/workspace/work# python -m torch2trt.test --name=ConvTranspose2d
| torch2trt.converters.ConvTranspose2d.test_1 | float32 | [(1, 3, 224, 224)] | {} | 4.77E-07 | 1.12e+04 | 842 | 0.119 | 1.18 |
[TensorRT] ERROR: Internal error: could not find any implementation for node (Unnamed Layer* 0) [Deconvolution], try increasing the workspace size with IBuilder::setMaxWorkspaceSize()
[TensorRT] ERROR: ../builder/tacticOptimizer.cpp (1461) - OutOfMemory Error in computeCosts: 0
| torch2trt.converters.ConvTranspose2d.test_2 | float32 | [(1, 3, 224, 224)] | {} | N/A | N/A | N/A | N/A | N/A |
| torch2trt.converters.ConvTranspose2d.test_square_kernel_equal_stride_mode | float32 | [(1, 3, 224, 224)] | {} | 3.58E-07 | 1.73e+04 | 600 | 0.0765 | 1.62 |
| torch2trt.converters.ConvTranspose2d.test_square_kernel_equal_stride_mode_unequal_op_size | float32 | [(1, 3, 224, 224)] | {} | 3.58E-07 | 1.11e+04 | 544 | 0.117 | 1.77 |
| torch2trt.converters.ConvTranspose2d.test_unequal_stride_mode | float32 | [(1, 3, 224, 224)] | {} | 2.38E-07 | 1.56e+04 | 187 | 0.0647 | 5.41 |
| torch2trt.converters.ConvTranspose2d.test_kernelsize_4 | float32 | [(1, 3, 7, 7)] | {} | 0.00E+00 | 2.32e+04 | 1.58e+04 | 0.0583 | 0.0753 |
| torch2trt.converters.ConvTranspose2d.test_kernelsize_4 | float32 | [(1, 3, 112, 112)] | {} | 3.58E-07 | 1.82e+04 | 1.95e+03 | 0.056 | 0.518 |
```

I ran the following command:

`python -m torch2trt.test --name=ConvTranspose2d`

and the unit test failing is for the following unit test

```
def test_2():
    return torch.nn.ConvTranspose2d(
        3, 6, 3, stride=(1, 2), padding=1, output_padding=(0, 1)
    )
```

### CODE REVIEW

**Comment by SrivastavaKshitij** [comment]
In `ConvTranspose2d.py` , If I change the following lines from :

```
if output_padding[0] + output_padding[1] > 0:
        layer.pre_padding = padding
        layer.post_padding = trt.tensorrt.DimsHW(
            padding[0] - output_padding[0], padding[1] - output_padding[1]
        )
    else:
        layer.padding = padding
```

to

```
if output_padding:
        layer.pre_padding = trt.tensorrt.DimsHW(padding)
        layer.post_padding = trt.tensorrt.DimsHW(output_padding)
    else:
        layer.padding = padding
```

5 out of 8 unit tests fail

```
root@7f9f14f6c747:/workspace/work# python -m torch2trt.test --name=ConvTranspose2d
<class 'tuple'> (0, 0)
| torch2trt.converters.ConvTranspose2d.test_1 | float32 | [(1, 3, 224, 224)] | {} | 3.58E-07 | 8.5e+03 | 741 | 0.111 | 1.05 |
<class 'tuple'> (0, 0)
[TensorRT] ERROR: Internal error: could not find any implementation for node (Unnamed Layer* 0) [Deconvolution], try increasing the workspace size with IBuilder::setMaxWorkspaceSize()
[TensorRT] ERROR: ../builder/tacticOptimizer.cpp (1461) - OutOfMemory Error in computeCosts: 0
| torch2trt.converters.ConvTranspose2d.test_only_padding | float32 | [(1, 3, 224, 224)] | {} | N/A | N/A | N/A | N/A | N/A |
<class 'tuple'> (0, 1)
[TensorRT] ERROR: Internal error: could not find any implementation for node (Unnamed Layer* 0) [Deconvolution], try increasing the workspace size with IBuilder::setMaxWorkspaceSize()
[TensorRT] ERROR: ../builder/tacticOptimizer.cpp (1461) - OutOfMemory Error in computeCosts: 0
| torch2trt.converters.ConvTranspose2d.test_2 | float32 | [(1, 3, 224, 224)] | {} | N/A | N/A | N/A | N/A | N/A |
<class 'tuple'> (0, 0)
| torch2trt.converters.ConvTranspose2d.test_square_kernel_equal_stride_mode | float32 | [(1, 3, 224, 224)] | {} | 3.58E-07 | 1.81e+04 | 586 | 0.0767 | 1.63 |
<class 'tuple'> (0, 0)
| torch2trt.converters.ConvTranspose2d.test_square_kernel_equal_stride_mode_unequal_op_size | float32 | [(1, 3, 224, 224)] | {} | 2.38E-07 | 1.09e+04 | 574 | 0.118 | 1.68 |
<class 'tuple'> (0, 0)
[TensorRT] ERROR: Internal error: could not find any implementation for node (Unnamed Layer* 0) [Deconvolution], try increasing the workspace size with IBuilder::setMaxWorkspaceSize()
[TensorRT] ERROR: ../builder/tacticOptimizer.cpp (1461) - OutOfMemory Error in computeCosts: 0
| torch2trt.converters.ConvTranspose2d.test_unequal_stride_mode | float32 | [(1, 3, 224, 224)] | {} | N/A | N/A | N/A | N/A | N/A |
<class 'tuple'> (0, 0)
| torch2trt.converters.ConvTranspose2d.test_kernelsize_4 | float32 | [(1, 3, 7, 7)] | {} | N/A | N/A | N/A | N/A | N/A |
<class 'tuple'> (0, 0)
[TensorRT] ERROR: Internal error: could not find any implementation for node (Unnamed Layer* 0) [Deconvolution], try increasing the workspace size with IBuilder::setMaxWorkspaceSize()
[TensorRT] ERROR: ../builder/tacticOptimizer.cpp (1461) - OutOfMemory Error in computeCosts: 0
| torch2trt.converters.ConvTranspose2d.test_kernelsize_4 | float32 | [(1, 3, 112, 112)] | {} | N/A | N/A | N/A | N/A | N/A |
NUM_TESTS: 8
NUM_SUCCESSFUL_CONVERSION: 3
NUM_FAILED_CONVERSION: 5
NUM_ABOVE_TOLERANCE: 0

```

**PS**: Disregard `<class 'tuple'> (0, 0)` , I had some print some statements which are not present in this PR.

**Comment by SrivastavaKshitij** [comment]
Previously , ConvTranspose2d didnt have any unit test capturing `padding , pre_padding or output_padding` test cases. 

I am short on ideas on how to fix the problem but I have narrowed down the problem to a combination of `padding & output padding`

**Comment by ShangxuanWu** [comment]
Yeah, we are suspecting this issue causing a vastly different output between TRT-ed and raw PyTorch model which uses `ConvTranspose2D`.

**Comment by sudeepf** [comment]
@ShangxuanWu @SrivastavaKshitij  do we know if the difference in the output is because of the output-padding parameter ?

**Comment by SrivastavaKshitij** [comment]
@sudeepf : yes, thats why the unit test containing `output padding` parameter  failed

**Comment by SrivastavaKshitij** [comment]
Potential solution is to adjust output size by playing around with other parameters other than output padding such as the following

Screen Shot 2020-07-21 at 5 04 06 PM
